### ISSUE TITLE
Misleading error when attempting to use `macro_rules` as an attribute

### ISSUE THREAD

**Issue by kpreid** [rust-lang/rust#132928]
### Code

Code provided by Discord user baltdev on the Rust Programming Language Discord server ([message link](https://discord.com/channels/442252698964721669/443150878111694848/1305685566880350218)).

```Rust
macro_rules! sample { () => {} }

#[sample]
pub struct S {}
```

### Current output

```Shell
error: cannot find attribute `sample` in this scope
 --> src/lib.rs:3:3
  |
3 | #[sample]
  |   ^^^^^^ consider moving the definition of `sample` before this call
  |
note: a macro with the same name exists, but it appears later at here
 --> src/lib.rs:1:14
  |
1 | macro_rules! sample { () => {} }
  |              ^^^^^^

warning: unused macro definition: `sample`
 --> src/lib.rs:1:14
  |
1 | macro_rules! sample { () => {} }
  |              ^^^^^^
  |
  = note: `#[warn(unused_macros)]` on by default
```

### Desired output

```Shell
error: `sample` is not an attribute macro
help: `macro_rules!` macros cannot be used as attribute macros
```

### Rationale and extra context

The error message is about a missing definition that is not missing, and incorrectly suggests that reordering the code might help. It should instead explain the fundamental problem with what is being attempted.

This issue is moot if https://github.com/rust-lang/rfcs/pull/3697 is implemented.

### Other cases

```Rust

```

### Rust Version

```Shell
1.82.0
```

**Comment by joshtriplett** [comment]
@kpreid It's not moot when RFC 3697 gets implemented. The error message is still misleading and should be improved. It'll just be able to say that the macro doesn't have any `attr` rules, rather than that this isn't possible.
